app/presenters/supergroups/supergroup.rb

```diff
@@ -0,0 +1,45 @@
+module Supergroups
+  class Supergroup
+  private
+
+    def format_document_data(documents, data_category: nil, include_timestamp: true)
+      documents.each.with_index(1).map do |document, index|
+        data = {
+          link: {
+            text: document["title"],
+            path: document["link"],
+            data_attributes: data_attributes(document["link"], document["title"], index)
+          },
+          metadata: {
+            document_type: document["content_store_document_type"].humanize
+          }
+        }
+
+        if include_timestamp && document["public_timestamp"]
+          data[:metadata][:public_updated_at] = Date.parse(document["public_timestamp"])
+        end
+
+        if data_category.present?
+          data[:link][:data_attributes][:track_category] = data_module_label + data_category
+        end
+
+        data
+      end
+    end
+
+    def data_attributes(base_path, link_text, index)
+      {
+        track_category: data_module_label + "DocumentListClicked",
```

Comment:
Looks like some of the tracking stuff won't work because the document list component doesn't have `data-module="track-click"`, it requires you to wrap the document-list in an element with that attribute. Not sure how I didn't spot that with the services work, sorry! That'll need changing there too.

In the long term we probably just want to update the component to do this by default if someone passes tracking info, but it will require updating the other frontend apps.

Reply:
Boo! 👎

Reply:
I've wrapped the partial in the data-module now and it seems to work OK.